Meadowlark-js/tests/e2e/scenarios/QueryStringValidation.test.ts

```diff
@@ -0,0 +1,287 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { createSchoolsInBulk, deleteListOfResources } from '../functions/BulkCreation';
+import { Clients, getAccessToken } from '../functions/Credentials';
+import { createResource, deleteResourceByLocation } from '../functions/Resources';
+import { baseURLRequest } from '../Setup';
+
+describe('When retrieving information', () => {
+  describe("given there's no data", () => {
+    it('should return the total count', async () => {
+      await baseURLRequest
+        .get('/v3.3b/ed-fi/schools')
+        .auth(await getAccessToken(Clients.Vendor1), { type: 'bearer' })
+        .expect(200)
+        .then((response) => {
+          expect(response.headers['total-count']).toEqual('0');
+          expect(response.body).toMatchInlineSnapshot(`[]`);
+        });
+    });
+  });
+
+  describe('when querying with filters', () => {
+    const total = 10;
+    let schools: Array<string>;
+
+    beforeAll(async () => {
+      const bulkCreation = await createSchoolsInBulk(total);
+      if (bulkCreation.errors) {
+        throw new Error('Error creating schools');
+      }
+      schools = bulkCreation.resources;
+    });
+
+    describe('when querying with limit', () => {
+      describe('when getting less than total', () => {
+        // Use the assessment1 credentials to bypass additional validations
+        it('should return the total and a subset of the results', async () => {
+          const limit = total - 1;
+          await baseURLRequest
+            .get(`/v3.3b/ed-fi/schools?limit=${limit}`)
+            .auth(await getAccessToken(Clients.Assessment1), { type: 'bearer' })
+            .expect(200)
+            .then((response) => {
+              expect(+response.headers['total-count']).toEqual(total);
+              expect(response.body.length).toEqual(limit);
+            });
+        });
+      });
+
+      describe('when requesting with invalid values', () => {
+        // Use the assessment1 credentials to bypass additional validations
+        it.each([0, 'zero'])('limit = %s', async (limit) => {
```

Comment:
Can we add a few injection / attack inputs?

```suggestion
        it.each([0, 'zero', '5; select * from users', '0)', '1%27'])('limit = %s', async (limit) => {
```